### PR TITLE
#160237207 Implement add vendor functionality

### PR DESCRIPTION
**Pull request by chibuezeayogu**
**What does the PR do?**
- Implements add vendor functionality

**Description of Task to be completed?**
- Ensures an admin can add a vendor

**How should this be manually tested?**
- Clone the repository [here](https://github.com/andela/AndelaEatsUI.git) and
-  cd into AndelaEatsUI and checkout to this branch `ft-implement-create-vendor-functionality-160237207`
- run `npm run bundle` and `npm run dev`
- Log in as an `Admin`, navigate to `admin dashboard`, select `Vendors`, and click on `Add Vendor`
- supply the required information and click `Add vendor button`
- run `npm test` to ensure all tests are passing

**What are the relevant pivotal tracker stories?**
- [#160237207](https://www.pivotaltracker.com/story/show/160237207)

**Screenshots**
- N/A

### CODE REVIEW

**Comment by abdulfataiaka** [comment]
Looks good to me.

File: client/src/styles/components/_loader.scss [hunk 1]
```diff
@@ -20,3 +21,8 @@
 @keyframes spin {
 	100%{ transform: rotate(360deg); }
 }
+
+.modal-loader {
+  position: relative;
+  margin: 0 auto;
+}
```
Comment (monsieurkayode) [inline client/src/styles/components/_loader.scss]: Add new line here

File: client/src/tests/components/Admin/Vendors/AddVendorModal.test.js [hunk 2]
```diff
@@ -40,4 +43,23 @@ describe('AddVendorModal Component', () => {
...
+    expect(clearErrorsSpy).toHaveBeenCalled();
+  });
+
+  it('should call formValidation method', () => {
+    const formValidationSpy = jest.spyOn(wrapper.instance(), 'formValidation');
+    const event = { preventDefault: jest.fn() };
+    wrapper.instance().formValidation(event);
+    expect(formValidationSpy).toHaveBeenCalled();
+  });
 });
```
Comment (monsieurkayode) [inline client/src/tests/components/Admin/Vendors/AddVendorModal.test.js]: Add new line here

File: client/src/tests/helpers/inputValidation.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+const FormField = {
+  lastName: "Chibueze",
+  firstName: "Ayogu",
+  errors: {}
+};
+
+test('Validate form field', () => {
+  expect(inputValidation(FormField)).toEqual({ isEmpty: true, errors: { } });
+});
```
Comment (monsieurkayode) [inline client/src/tests/helpers/inputValidation.test.js]: Add new line here

File: client/src/tests/actions/vendorsAction.test.js [hunk 4]
```diff
@@ -4,12 +4,17 @@ import {
...
+  CREATE_VENDOR_FAILURE,
+  CREATE_VENDOR_LOADING
 } from '../../actions/actionTypes';
 import {
   baseUrl,
   fetchVendors,
+  createVendor
 } from '../../actions/vendorsAction';
 import vendors from '../__mocks__/mockVendors';
+import { newVendor, createdVendor } from '../__mocks__/mockNewVendor';
```
Comment (monsieurkayode) [inline client/src/tests/actions/vendorsAction.test.js]: Separate import properties by new lines as done on line 11

**Comment by chibuezeayogu** [comment]
@monsieurkayode All concerns have been addressed

**Pull request merged**